base/reduce.jl

```diff
@@ -131,7 +131,13 @@ mapfoldr(f, op, v0, itr) = mapfoldr_impl(f, op, v0, itr, endof(itr))
 Like `mapfoldr(f, op, v0, itr)`, but using the first element of `itr` as `v0`. In general,
 this cannot be used with empty collections (see `reduce(op, itr)`).
 """
-mapfoldr(f, op, itr) = (i = endof(itr); mapfoldr_impl(f, op, f(itr[i]), itr, i-1))
+function mapfoldr(f, op, itr)
+    i = endof(itr)
+    if i == 0
```

Comment:
I wondered the same given [the `mapfoldl` implementation](https://github.com/JuliaLang/julia/blob/205a8cdac7bd2a3f8cf31133b78a169ed74a0f59/base/reduce.jl#L66) takes what might be a more general approach. Thoughts?